tests/e2e/testnet/genesis.json

```diff
@@ -0,0 +1,298 @@
+{
+  "height": 1,
+  "genesis_time": "2022-09-26T04:06:00.331342488Z",
+  "chain_id": "test: oasis-core tests",
+  "registry": {
+    "params": {
+      "debug_allow_unroutable_addresses": true,
+      "debug_allow_test_runtimes": true,
+      "gas_costs": {
+        "deregister_entity": 1000,
+        "register_entity": 1000,
+        "register_node": 1000,
+        "register_runtime": 1000,
+        "runtime_epoch_maintenance": 1000,
+        "unfreeze_node": 1000,
+        "update_keymanager": 1000
+      },
+      "max_node_expiration": 5,
+      "enable_runtime_governance_models": {
+        "entity": true,
+        "runtime": true
+      }
+    },
+    "entities": [
+      {
+        "untrusted_raw_value": "o2F2AmJpZFggJTUtHd4XYQjh//e6eYU7Pa/XMFG88WE+jixvceIfWrllbm9kZXOBWCAtC7hm0WDw4nQwLgzhAx5RHsizpe3gD8Jb48r/tM+IfQ==",
+        "signature": {
+          "public_key": "JTUtHd4XYQjh//e6eYU7Pa/XMFG88WE+jixvceIfWrk=",
+          "signature": "L9wqH/IjJ3AdgledgK/1qU86f5kKjWy/zpd3cS8YkOYZyQBi+z98wNANy6ACiW3kpAD5uI/qcTg/ez+nE7dJCw=="
+        }
+      },
+      {
+        "untrusted_raw_value": "omF2AmJpZFgg+MJpnSTzc11dNI5emMa+asCJH5cxBiBCcpbYE4XBdso=",
+        "signature": {
+          "public_key": "+MJpnSTzc11dNI5emMa+asCJH5cxBiBCcpbYE4XBdso=",
+          "signature": "LXp4Fl89LDJyqHR92PWfIN5yod+eOwZBfresKfiEoMReuZugzbIFjMuBrC8ruazd/UAbfU6r6MPIM4H6YQOvCA=="
+        }
+      },
+      {
+        "untrusted_raw_value": "omF2AmJpZFggHc0T6ypN7Ytv3t+n5LyJjCd93geUMTo82BR8iS1sRkY=",
+        "signature": {
+          "public_key": "Hc0T6ypN7Ytv3t+n5LyJjCd93geUMTo82BR8iS1sRkY=",
+          "signature": "IRxC/BtxNGVjN21fzNGu/rNm1FN0wTwdrffEUJzigPOuSaP6dxNtxxKYWm89AMIByc29g2qtv33UPImDwH0lDQ=="
+        }
+      },
+      {
+        "untrusted_raw_value": "omF2AmJpZFggBTDHzkZYnKPQKUcnN4ieYpLBKRLMu/tmktwpbLduAj8=",
+        "signature": {
+          "public_key": "BTDHzkZYnKPQKUcnN4ieYpLBKRLMu/tmktwpbLduAj8=",
+          "signature": "VDTt/fqiGFjFYUvAr7ar2LMSzlb5FEQnjzpJYrgUlhdFKukJI3cyCu0RApy9d4LM+eiv3L8uj5nmHLItovX2Aw=="
+        }
+      },
+      {
+        "untrusted_raw_value": "omF2AmJpZFggkNGfL2S9wpni8G2phHcbUYRmnwSsZFny90VFTMXvDKo=",
+        "signature": {
+          "public_key": "kNGfL2S9wpni8G2phHcbUYRmnwSsZFny90VFTMXvDKo=",
+          "signature": "Dpa4leeFoFLd2OK5AsxqVghSivB8p2dibzVERJgODzGZDcoFpTurTX2g+MuIEoIP4TjWWr432l4i9lW1nniJAg=="
+        }
+      },
+      {
+        "untrusted_raw_value": "omF2AmJpZFggTqUyj5Q+9vZtqu10yw6Zw7HEX3Ywe0JQA9vHyzY47TU=",
+        "signature": {
+          "public_key": "TqUyj5Q+9vZtqu10yw6Zw7HEX3Ywe0JQA9vHyzY47TU=",
+          "signature": "UiuPWSjYdXih3iEGBTPyWTWoXb4cXfPQi4xnrlgooXpin876uZe4Uy3D5tLYrqDCGpDJ+r/8r0gTJ6aIwCuvBQ=="
+        }
+      }
+    ],
+    "nodes": [
```

Comment:
Do we need to insert the genesis entities, nodes etc into the DB? During a normal run, genesis state is inserted into the DB via the manually-run migrator. Maybe we need to do it _in principle_, but the lack of that data in the DB doesn't affect the current e2e tests?

(Vaguely related: I'm no fan of the migrator-generated SQL files and would prefer to convert the genesis file to SQL on the fly during migrations. Checking autogenerated content into git is usually a bad idea. If we were to do the conversion on the fly, the whole thing would also be amenable to e2e testing. Do you or @ennsharma recall a reason for the files? If not, sidestepping the files is a low-priority issue to file.)

Reply:
> Do we need to insert the genesis entities, nodes etc into the DB?

That is the right thing to do.

> the lack of that data in the DB doesn't affect the current e2e tests?

Correct. We don't make those assertions right now.

> If we were to do the conversion on the fly, the whole thing would also be amenable to e2e testing.

I'm open to this. I did like seeing the migrations though for debugging purposes, but perhaps that is a problem to be solved there.

Reply:
Thanks for the explanation. Filed https://github.com/oasisprotocol/oasis-indexer/issues/153 to track. Mind adding a TODO comment in the code here, referencing #153?

Reply:
Cool. I don't think I can add a comment on this line, but will add a comment in the analyzer code.